RFS/src/main/java/com/rfs/worker/DocumentsRunner.java

```diff
@@ -0,0 +1,63 @@
+package com.rfs.worker;
+
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.rfs.cms.CmsClient;
+import com.rfs.cms.CmsEntry;
+import com.rfs.common.DocumentReindexer;
+import com.rfs.common.IndexMetadata;
+import com.rfs.common.LuceneDocumentsReader;
+import com.rfs.common.ShardMetadata;
+import com.rfs.common.SnapshotShardUnpacker;
+
+public class DocumentsRunner implements Runner {
+    private static final Logger logger = LogManager.getLogger(DocumentsRunner.class);
+
+    private final DocumentsStep.SharedMembers members;
+
+    public DocumentsRunner(GlobalState globalState, CmsClient cmsClient, String snapshotName, IndexMetadata.Factory metadataFactory,
+                ShardMetadata.Factory shardMetadataFactory, SnapshotShardUnpacker unpacker, LuceneDocumentsReader reader,
+                DocumentReindexer reindexer) {
+        this.members = new DocumentsStep.SharedMembers(globalState, cmsClient, snapshotName, metadataFactory, shardMetadataFactory, unpacker, reader, reindexer);
+    }
```

Comment:
Oh, interesting - so are you thinking that the SharedMembers are a higher level abstraction rather than something just internal to the DocumentStep classes?

Reply:
Yea I think so - since most members are not stateful so it seems like they could be part of a higher level construction for reuse.

And for Global state, I think I would recommend just removing it, seems like it represents a 'higher' step in, but the downstream systems don't seem to need to know where they are at save for a couple of log messages.

Reply:
The Steps set the Phase and Work Item in the GlobalState.  These aren't used currently but have future uses per the design.